pep-0654.rst

```diff
@@ -318,34 +318,61 @@ typically either query to check if it has leaf exceptions for which some
 condition holds (using ``subgroup`` or ``split``) or format the exception
 (using the ``traceback`` module's methods).
 
-It is unlikely to be useful to inspect the individual leaf exceptions. To see
-why, suppose that an application caught an ``ExceptionGroup`` raised in an
-``asyncio.gather()`` call. At this stage, the context for each specific
+It is less likely to be useful to iterate over the individual leaf exceptions.
+To see why, suppose that an application caught an ``ExceptionGroup`` raised in
+an ``asyncio.gather()`` call. At this stage, the context for each specific
 exception is lost. Any recovery for this exception should have been performed
 before it was grouped with other exceptions into the ``ExceptionGroup`` [10]_.
 Furthermore, the application is likely to react in the same way to any number
 of instances of a certain exception type, so it is more likely that we will
 want to know whether ``eg.subgroup(T)`` is None or not, than we are to be
 intersted in the number of ``Ts`` in ``eg``.
 
-If it does turn out to be necessary for an applicaiton to iterate over the
-individual exceptions of an ``ExceptionGroup`` ``eg``, this can be done by
-calling ``traverse(eg)``, where ``traverse`` is defined as follows:
+However, there are situations where it is necessary to inspect the
+individual leaf exceptions. For example, suppose that we have an
+``ExceptionGroup`` ``eg`` and we want to log the ``OSErrors`` that have a
+specific error code and reraise everything else. We can do this with
+``subgroup`` as follows:
 
 .. code-block::
 
-    def traverse(exc, tbs=None):
+    def log_and_ignore_ENOENT(err):
+        if isinstance(err, OSError) and err.errno == ENOENT:
+            log(err)
+            return False
+        else:
+            return True
+
+    try:
+        . . .
+    except ExceptionGroup as eg:
+        eg = eg.subgroup(log_and_ignore_ENOENT)
+        if eg is not None:
+            raise eg
+
+
+In the previous example, when ``log_and_ignore_ENOENT`` is invoked on a leaf
+exception, only part of this exception's traceback is accessible - the part
+referenced from its ``__traceback__`` field. If we need the full traceback,
+we need to look at the concatenation of the tracebacks of the exceptions on
+the path from the root to this leaf. we can get that with direct iteration,
+recursively, as follows:
+
+.. code-block::
+
+    def leaf_generator(exc, tbs=None):
```

Comment:
Maybe we could show a little code fragment that uses this as well? It would have to include at least
```
try:
    . . .
except *OSError as eg:
    for exc, tbs in leaf_generator(eg):
        if exc.errno != ENOENT:
            print(exc)
            <show how to print the traceback using the traceback module>
```
I think the last line would have to call extract_tb() for each item of tbs, concatenate the lists (in which order?), and then call print_list() on the concatenation. I'm too lazy to research how exactly.